pkg/util/grpcclient/instrumentation.go

```diff
@@ -7,11 +7,13 @@ import (
 	"github.com/weaveworks/common/middleware"
 	"google.golang.org/grpc"
 
+	"github.com/cortexproject/cortex/pkg/util/grpcutil"
 	cortexmiddleware "github.com/cortexproject/cortex/pkg/util/middleware"
 )
 
 func Instrument(requestDuration *prometheus.HistogramVec) ([]grpc.UnaryClientInterceptor, []grpc.StreamClientInterceptor) {
 	return []grpc.UnaryClientInterceptor{
+			grpcutil.HTTPHeaderPropagationClientInterceptor(),
 			otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer()),
 			middleware.ClientUserHeaderInterceptor,
 			cortexmiddleware.PrometheusGRPCUnaryInstrumentation(requestDuration),
```

Comment:
Should we also have the middleware for the stream apis? (QueryStream for instance) (StreamClientInterceptor)